### DISCUSSION TITLE
New to Lowdefy. Shall I wait for v4, before I start some serious development?

### DISCUSSION THREAD

**Discussion by samyakbhuta** [lowdefy/lowdefy#1100] [Questions and Help] [answered]
### Background
* I did basic tutorial and find Lowdefy very promising.
* I want to give it a try for a application idea that I have in mind.
* I decided to dive into the docs/issues and found out that v4 will have lot of changes. `_ref` : #956

### Question
* When is v4 coming? Is it around the corner?
* Shall I start developing my app in v4 or I just start with the v3.x - what is recommended? 
* I am assuming that starting with v4 would have more difficulties in the beginning and over time, my app will just grow along with the v4. On the other hand, if I start with v3, I will benefit from existing stable code and code examples but will have to devote time to refactor my app to adjust it to v4. Is that correct assumption?

**Comment by SamTolmay** [comment, marked as answer]
Hi @samyakbhuta 

I am glad you find the project promising, and hope it will be useful for you.

The changes for version 4 do seem like a lot, but if you are not using features like custom blocks, custom JsActions and the custom _js operator, it won't be such a large job to convert to version 4. Since the configuration is declarative, we can make large changes to the underlying framework, without introducing major breaking changes to the configuration schema.

The one change you will need to make is to convert requests to use payloads on requests. This does mean you will need to change all your requests, but we expect this to be a relatively easy fix, especially if your app is not very complicated yet.

We are planning on converting some of our production apps to version 4 by the end of February, and having a release candidate release in early March. This should be reasonably stable, but we will be working on documentation at that time, so there won't be a lot of documentation on the new features, which we will have ready when we release the stable version.

I think it would be best to start using version 3 and then convert to version 4 once we have released a release candidate or stable version.

**Comment by samyakbhuta** [comment]
@SamTolmay thanks for the detailed reply :pray:
